report/src/main/webapp/app/services/metadata.service.js

```diff
@@ -149,6 +152,85 @@ define(['angularAMD', 'metadataCacheService', 'metadataEndpointService'],
 
       }
 
+      function ComparatorStepResultsWrapper(comparatorStepResults) {
+        var stepResults = comparatorStepResults,
+            resultsWrapper = {
+              getStepResults: getStepResults,
+              getStatus: getStatus,
+              setStatus: setStatus,
+              isAcceptable: isAcceptable,
+              isRebaseable: isRebaseable
+            };
+
+        return resultsWrapper;
+
+        function getStepResults() {
+          return stepResults;
+        }
+
+        function getStatus() {
+          var statusesBySeverity = [
+            'PROCESSING_ERROR',
+            'REBASED',
+            'FAILED',
+            'WARNING',
+            'CONDITIONALLY_PASSED',
+            'PASSED'];
+
+          var foundStatus;
+
+          _.forEach(statusesBySeverity, function (status) {
+            if (!foundStatus && anyResultHasStatus(status)) {
+              foundStatus = status;
+            }
+          });
+
+          return foundStatus;
+        }
+
+        function anyResultHasStatus(status) {
+          var hasStatus = false;
+
+          _.forEach(stepResults, function (stepResult) {
+            if (stepResult.status === status) {
+              hasStatus = true;
+            }
+          });
+
+          return hasStatus;
+        }
+
+        function setStatus(newStatus) {
+          _.forEach(stepResults, function (stepResult) {
+            stepResult.status = newStatus;
+          });
+        }
+
+        function isAcceptable() {
+          var isAnyAcceptable = false;
+
+          _.forEach(stepResults, function (stepResult) {
+            if (stepResult.acceptable) {
+              isAnyAcceptable = true;
+            }
+          });
+
+          return isAnyAcceptable;
+        }
+
+        function isRebaseable() {
+          var isAnyRebaseable = false;
+
+          _.forEach(stepResults, function (stepResult) {
```

Comment:
You don't need to iterate to the end of the loop. You need to find first `stepResult` that is rebaseable.
Try to use `find` insead of `forEach`:

```js
var rebaseableStepResult = _.find(stepResults, function (stepResult) {
  return stepResult.rebaseable;
});
return !!rebaseableStepResult;
```

Reply:
Thanks! I fixed the code, also in other places you mentioned too.